streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/TaskMetrics.java

```diff
@@ -0,0 +1,263 @@
+            metricName,
+            descriptionOfRate,
+            descriptionOfCount
+        );
+        return sensor;
+    }
+
+    private static Sensor avgAndMaxSensor(final String threadId,
+                                          final String taskId,
+                                          final String metricName,
+                                          final String descriptionOfAvg,
+                                          final String descriptionOfMax,
+                                          final RecordingLevel recordingLevel,
+                                          final StreamsMetricsImpl streamsMetrics,
+                                          final Sensor... parentSensors) {
+        final Sensor sensor = streamsMetrics.taskLevelSensor(threadId, taskId, metricName, recordingLevel, parentSensors);
+        final Map<String, String> tagMap = streamsMetrics.taskLevelTagMap(threadId, taskId);
+        addAvgAndMaxToSensor(
+            sensor,
+            TASK_LEVEL_GROUP,
+            tagMap,
+            metricName,
+            descriptionOfAvg,
+            descriptionOfMax
+        );
+        return sensor;
+    }
+
+    private static Sensor invocationRateAndCountAndAvgAndMaxLatencySensor(final String threadId,
+                                                                          final String taskId,
+                                                                          final String metricName,
+                                                                          final String descriptionOfRate,
+                                                                          final String descriptionOfCount,
+                                                                          final String descriptionOfAvg,
+                                                                          final String descriptionOfMax,
+                                                                          final RecordingLevel recordingLevel,
+                                                                          final StreamsMetricsImpl streamsMetrics,
+                                                                          final Sensor... parentSensors) {
+        final Sensor sensor = streamsMetrics.taskLevelSensor(threadId, taskId, metricName, recordingLevel, parentSensors);
+        final Map<String, String> tagMap = streamsMetrics.taskLevelTagMap(threadId, taskId);
+        addAvgAndMaxToSensor(
+            sensor,
+            TASK_LEVEL_GROUP,
+            tagMap,
+            metricName + LATENCY_SUFFIX,
+            descriptionOfAvg,
+            descriptionOfMax
+        );
+        addInvocationRateAndCountToSensor(
+            sensor,
+            TASK_LEVEL_GROUP,
+            tagMap,
+            metricName,
+            descriptionOfRate,
+            descriptionOfCount
+        );
+        return sensor;
+    }
+
+    private static Sensor emptySensor(final String threadId,
```

Comment:
What does the function name mean?

Reply:
An empty sensor is a sensor without added metrics. I need such an empty sensor for sensors that exist in one built-in metrics version but not in the other. Examples are the parent commit sensor on thread-level that exist in `0100-24` but not in `latest` or the punctuate sensor on task-level which exist in `lasest` but not in `0100-24`. In such a way I do not need to use `Optional` and change the code to record the metrics.

Reply:
Got it, I'm wondering if it would add any unnecessary overhead to the metrics registry with the empty sensors compared with the `optional` approach? Maybe I'm paranoid here, just double checking to make sure that the access pattern into the registry would not be largely impacted by the larger map with empty sensors.

Reply:
I realized by working on processor node metrics, that the sensors that I created as empty sensors on task-level like `punctuateSensor` are used as parent sensors on processor node level for version `0100-24`. So they will not stay empty. So I guess in the end there will only be a few empty sensors. If we experience performance issue we can still rewrite the empty sensor with the `Optional` approach.

Reply:
I guess if you want to make it really work the way you were planning, you could extract an interface for Sensor and then declare an EmptySensor that just discards updates from children... not sure if you want to potentially get into a mess, though.